extension/background/intentRunner.js

```diff
@@ -531,6 +531,42 @@ export function getRegisteredNicknames() {
   return registeredNicknames;
 }
 
+const registeredPageName = {};
+
+export function registerPageName(name, context) {
+  name = name.toLowerCase();
+  if (!context) {
+    delete registeredPageName[name];
+    log.info("Removed nickname for page", name);
+  } else {
+    registeredPageName[name] = context;
+    log.info(
+      "Added nickname for page",
+      name,
+      "->",
+      context.name,
+      context.slots
+    );
+  }
+  browser.storage.sync.set({ registeredPageName });
+}
+
+async function initRegisteredPageName() {
+  const result = await browser.storage.sync.get(["registeredPageName"]);
+  if (result.registeredPageName) {
+    for (const name in result.registeredPageName) {
+      const value = result.registeredPageName[name];
+      const context = new IntentContext(value);
+      registeredPageName[name] = context;
+      log.info("Loaded page name", name, context.name, context.slots);
+    }
+  }
+}
+
+export function getRegisteredPageName() {
```

Comment:
This should be like `getRegisteredPageName(name)` – and return null if there's no page with that name.